.github/workflows/dev-cmake.yml

```diff
@@ -0,0 +1,346 @@
+
+      - name: check clang version
+        shell: bash
+        run: |
+          which clang
+          clang -v
+
+      - name: check cmake version
+        shell: bash
+        run: |
+          which cmake
+          cmake --version
+
+      - name: Set file base name (MacOS_latest)
+        id: set-file-base
+        run: |
+          FILE_NAME_BASE=$(echo "${{ inputs.file_base }}")
+          echo "FILE_BASE=$FILE_NAME_BASE" >> $GITHUB_OUTPUT
+          if [[ '${{ inputs.use_environ }}' == 'release' ]]
+          then
+            SOURCE_NAME_BASE=$(echo "${{ inputs.snap_name }}")
+          else
+            SOURCE_NAME_BASE=$(echo "hdfsrc")
+          fi
+          echo "SOURCE_BASE=$SOURCE_NAME_BASE" >> $GITHUB_OUTPUT
+
+      # Get files created by release script
+      - name: Get tgz-tarball (MacOS_latest)
+        uses: actions/download-artifact@cc203385981b70ca67e1cc392babf9cc229d5806 # v4.1.9
+        with:
+              name: tgz-tarball
+              path: ${{ github.workspace }}
+
+      - name: List files for the space (MacOS_latest)
+        run: |
+              ls ${{ github.workspace }}
+              ls ${{ runner.workspace }}
+
+      - name: Uncompress source (MacOS_latest)
+        run: tar -zxvf ${{ github.workspace }}/${{ steps.set-file-base.outputs.FILE_BASE }}.tar.gz
+
+      - name: CMake Configure
+        shell: bash
+        run: |
+          mkdir "${{ runner.workspace }}/build"
+          cd "${{ runner.workspace }}/build"
+          cmake -C ${{ github.workspace }}/hdfsrc/config/cmake/cacheinit.cmake \
+               -G Ninja \
+               -DCMAKE_BUILD_TYPE=Developer \
+               -DBUILD_SHARED_LIBS=ON \
+               -DHDF5_ENABLE_ALL_WARNINGS=ON \
+               -DHDF5_ENABLE_DEBUG_H5B2=ON \
+               -DHDF5_ENABLE_DEBUG_H5FS_ASSERT=ON \
+               -DHDF5_ENABLE_PARALLEL:BOOL=OFF \
+               -DHDF5_BUILD_CPP_LIB:BOOL=OFF \
+               -DHDF5_BUILD_FORTRAN=OFF \
+               -DHDF5_BUILD_JAVA=OFF \
+               -DHDF5_BUILD_DOC=OFF \
+               -DLIBAEC_USE_LOCALCONTENT=OFF \
+               -DZLIB_USE_LOCALCONTENT=OFF \
+               -DHDF_TEST_EXPRESS=0 \
+               ${{ github.workspace }}/hdfsrc
+
+      - name: CMake Build
+        run: cmake --build . --parallel 3 --config Developer
+        working-directory: ${{ runner.workspace }}/build
+
+      - name: CMake Run Tests
+        env:
+          HDF5TestExpress: 0
+        run: ctest . --parallel 2 -C Developer -V -R H5TESTXPR -E fheap
+        working-directory: ${{ runner.workspace }}/build
+
+  ####### clang builds
+  build_and_test_linux_clang:
+    # Linux (Ubuntu) w/ clang + CMake
+    #
+    name: "Ubuntu Clang CMake"
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Install CMake Dependencies (Linux_clang)
+        run: |
+          sudo apt-get update
+          sudo apt-get install ninja-build doxygen graphviz curl libtinfo5
+
+      - name: add clang to env
```

Comment:
This is fine for now I suppose, but we really should stop installing things that are already there and just use the defaults. Ubuntu has clang and installing a specific version here just adds maintenance for no reason. Same with forcing a particular version of GCC, installing and using Ninja (for some reason), etc. We're just moving our builds away from what most people are going to be using.